### PR TITLE
Ondrej-php PPA fixes/enhancements for Ubuntu

### PR DESCRIPTION
**Pull request by tdm4**
Hi,

  I've been doing work to get mayflower-php working with the ondrej-php PPA. It appears ondrej/php5 and ondrej/php5.6 are now deprecated and he has combined PHP 5.5, 5.6 and 7.0 into one repo. This completely breaks the module as it currently is. I've also put in some fixes so that it actually adds the PPA before it even tries to install the packages (in packages.pp, pear.pp and dev.pp).

  Please test and let me know if all works as it should. I've tested it on Ubuntu 14.04 (Trusty) with versions 5.5, 5.6, and 7.0 with the PPA.

### CODE REVIEW

**Comment by tdm4** [comment]
Oh I should note, this is what I've got in hiera:

`php::globals::php_version: '5.6'`
`php::manage_repos: true`
`php::fpm: true`
`php::dev: true`
`php::composer: true`
`php::pear: true`
`php::phpunit: false`

**Comment by tdm4** [comment]
I think the tests are incorrect because:
Failures:
  1) php::fpm on ubuntu-12.04-x86_64 when called with no parameters should contain Service[php5-fpm] with ensure => "running"

     Failure/Error:
       should contain_service('php5-fpm').with({
         'ensure' => 'running',
       })
       expected that the catalogue would contain Service[php5-fpm]
     # ./spec/classes/php_fpm_spec.rb:18:in `block (5 levels) in <top (required)>'

It looks like services are now handled by php::fpm::service

**Comment by tdm4** [comment]
I will work on fixing the tests as the pathnames and service names have changed for Ubuntu and using ondrej-php

**Comment by tdm4** [comment]
I've squashed my 19 commits into a single commit to make it easier.

**Comment by mmoll** [comment]
Thanks @tdm4, that's looking good! Could anybody test this and report back here?

**Comment by mmoll** [comment]
also, please rebase after the merge of #192

**Comment by mmoll** [comment]
a bit of documentation would also be nice :newspaper:

**Comment by tdm4** [comment]
@mmoll where's best to put documentation?

**Comment by mmoll** [comment]
@tdm4 in this case just in the README as further or updated examples

**Comment by tdm4** [comment]
@mmoll Do you think we could deprecate php::repo::ubuntu::ppa ? Seeing as how it dependent on version, and version is what people really would rather change rather than figuring out which repo to select?

**Comment by mmoll** [comment]
@tdm4 yes, that makes sense. However, I'd rather want to merge this PR very soon and then you could work on removing `php::repo::ubuntu::ppa`. As we're bumping the major version anyway, we don't need to care about compatibility. :boom:

**Comment by tdm4** [comment]
@mmoll sounds good to me! 👍

File: README.md [hunk 1]
```diff
@@ -189,10 +189,28 @@ upstream.
 
 To use an alternate PPA, Ondrej's PHP 5.6 for example, use the below hiera snippet
 ```yaml
-php::repo::ubuntu::ppa: 'ondrej/php5-5.6'
+php::repo::ubuntu::ppa: 'ondrej/php'
 php::manage_repos: true
 ```
 
+### Ubuntu systems and Ondrej's PPA
```
Comment (mmoll) [inline README.md]: nitpick: it's Ond**ř**ej (#79 for the background ;))

**Comment by mmoll** [comment]
merged, thanks @tdm4!

**Comment by tdm4** [comment]
@mmoll Great! Thanks for merging it!

**Pull request merged**